src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -191,8 +191,24 @@ public function rawUpdate(array $attributes = [])
      */
     public function getRelationExistenceCountQuery(Builder $query, Builder $parentQuery)
     {
+        return $this->getRelationExistenceAggregateQuery($query, $parentQuery, 'count', '*');
+    }
+
+    /**
+     * Add the constraints for a relationship aggregate query.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  \Illuminate\Database\Eloquent\Builder  $parentQuery
+     * @param  string  $function
+     * @param  string  $column
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function getRelationExistenceAggregateQuery(Builder $query, Builder $parentQuery, $function, $column)
+    {
+        $wrappedColumn = $query->getQuery()->getGrammar()->wrap($column);
+
         return $this->getRelationExistenceQuery(
-            $query, $parentQuery, new Expression('count(*)')
+            $query, $parentQuery, new Expression($function.'('.$wrappedColumn.')')
```

Comment:
Do you think there's any value in keeping a list of valid functions? So that we can tell the user that aggregate is not supported, otherwise I worry users might end up with domain code that works with one database engine and not another (which can lead to obscure bugs). It would be better that users be very explicit when they choose to use an aggregation that their particular database supports.

Reply:
I'd guess the database error message would be clear enough.

Reply:
Again, my concern is database compatibility – if you have to switch from one database to another for some reason, and your new database doesn't support the same aggregation (or worse, for some reason has a slightly different syntax), you'd be in a world of pain. I'm not sure this is anything that might happen, but I did want to bring it up just in case.

Reply:
Last year I was part of switching a mid-sized Laravel project from MySQL to Postgres.

I don't remember any Eloquent-related code paths had to be touched. Only MySQL specific SQL statements (💥 ).

To sum it up: this kind of thinking is very much appreciated!

Reply:
I agree with @deleugpn. What's the benefit of keeping a list of valid functions? In other words, how it would improve the user experience? 

Having a list of valid functions or not, the error only would be triggered when the query is executed

Reply:
@storrella if developers have a discrepancy between their production and dev environment, where they might run MySQL 8 locally and MySQL 5.6 in prod, they could run into issues. Those issues would be discovered at time of dev if you had a whitelist. Docker helps with this, but I don't think it's necessarily fair to assume a perfect dev setup either. Just feels like one of those things that's likely to go wrong when we let users input literally anything.

Reply:
Wouldn't we end up whitelisting both the MySQL 8 function as well as the MySQL 5.6 and be back to square 1? I guess like @storrella I'm having difficulties understanding the benefit of this as well since it's on Eloquent side. This isn't on a driver-specific Grammar class, so there's no point in whitelisting anything. Whatever error you get here you would get with your database as well. The difference is we would need to whitelist functions and map to a specific error message. Instead, let's not whitelist anything and let the database explain what went wrong.